### PR TITLE
tools: perf: document lib.figure.flat module

### PR DESCRIPTION
**Pull request by janekmi**
Requires:

- [x] #1358
- [x] #1361
- [x] #1364

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)

### CODE REVIEW

**Review by ldorau — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: 0 of 12 files reviewed, all discussions resolved (waiting on @ldorau)

File: tools/perf/lib/figure/flat.py [hunk 1]
```diff
@@ -4,7 +4,20 @@
...
 #
 
-"""flat.py -- helper flattening sets of objects (EXPERIMENTAL)"""
+#
+# flat.py
+#
+
+"""figure list flattening tools (EXPERIMENTAL)
+
+A 'flat' list a figures is a list where a single element on this list represents
```
Comment (ldorau) [inline tools/perf/lib/figure/flat.py:13]: Maybe:
A 'flat' list of figures is ... ?

**Review by grom72 — COMMENTED** [review]
@janekmi please rebase with #1358 and #1361

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366#-:-MpkQ9sG5EbAcshUc22z:bd30pot)* status: 0 of 12 files reviewed, 1 unresolved discussion (waiting on @grom72, @janekmi, and @ldorau)

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: 0 of 12 files reviewed, 1 unresolved discussion (waiting on @grom72 and @ldorau)

___
*[tools/perf/lib/figure/flat.py, line 13 at r1](https://reviewable.io/reviews/pmem/rpma/1366#-MpflcGB6RTTuFOJh47v-r1-13:-MpkfTgm0QH3w0BYLn_v:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/a781821124a6bfef8a56db23f682e60743f0ab16/tools/perf/lib/figure/flat.py#L13)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Maybe:
A 'flat' list of figures is ... ?
</blockquote></details>

Done.

**Review by ldorau — CHANGES_REQUESTED** [review]
> Reviewed 1 of 12 files at r1, 10 of 10 files at r2, 1 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: 12 of 13 files reviewed, 1 unresolved discussion (waiting on @grom72, @janekmi, and @ldorau)

___
*[tools/perf/lib/figure/flat.py, line 17 at r3](https://reviewable.io/reviews/pmem/rpma/1366#-MpkmsN_BNahMZQ4qjJq:-MpkmsN_BNahMZQ4qjJr:b-aih6db) ([raw file](https://github.com/pmem/rpma/blob/e277f650148a95b35155e8a4ee93b2c1e56c5662/tools/perf/lib/figure/flat.py#L17)):*
> ```
> figures are they are described
> ```

??? "figures described" ?

**Review by ldorau — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi)

___
*[tools/perf/lib/figure/flat.py, line 14 at r3](https://reviewable.io/reviews/pmem/rpma/1366#-MpknO4C3VeJAAq-584L:-MpknO4C3VeJAAq-584M:b-ykst13) ([raw file](https://github.com/pmem/rpma/blob/e277f650148a95b35155e8a4ee93b2c1e56c5662/tools/perf/lib/figure/flat.py#L14)):*
> ```
> represents exactly one output figure with all series representing
> ```

I do not understand that ("represents ... representing ... ")

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: 12 of 13 files reviewed, 2 unresolved discussions (waiting on @ldorau)

___
*[tools/perf/lib/figure/flat.py, line 14 at r3](https://reviewable.io/reviews/pmem/rpma/1366#-MpknO4C3VeJAAq-584L:-Mpksvhm9B0mMlLBa9Il:b-k6olmy) ([raw file](https://github.com/pmem/rpma/blob/e277f650148a95b35155e8a4ee93b2c1e56c5662/tools/perf/lib/figure/flat.py#L14)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

> ```
> represents exactly one output figure with all series representing
> ```

I do not understand that ("represents ... representing ... ")
</blockquote></details>

What do you think now?
___
*[tools/perf/lib/figure/flat.py, line 17 at r3](https://reviewable.io/reviews/pmem/rpma/1366#-MpkmsN_BNahMZQ4qjJq:-Mpksz250H_t2q5aDIFN:b-yw8ca2) ([raw file](https://github.com/pmem/rpma/blob/e277f650148a95b35155e8a4ee93b2c1e56c5662/tools/perf/lib/figure/flat.py#L17)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

> ```
> figures are they are described
> ```

??? "figures described" ?
</blockquote></details>

Any better?

**Review by ldorau — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

___
*[tools/perf/lib/figure/flat.py, line 14 at r3](https://reviewable.io/reviews/pmem/rpma/1366#-MpknO4C3VeJAAq-584L:-Mpkw6ubCoMTegS3XeEY:bgswke0) ([raw file](https://github.com/pmem/rpma/blob/e277f650148a95b35155e8a4ee93b2c1e56c5662/tools/perf/lib/figure/flat.py#L14)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

What do you think now?
</blockquote></details>

It is OK now
___
*[tools/perf/lib/figure/flat.py, line 17 at r3](https://reviewable.io/reviews/pmem/rpma/1366#-MpkmsN_BNahMZQ4qjJq:-MpkvcanB6GNUrfsaoTJ:b-1ctm1u) ([raw file](https://github.com/pmem/rpma/blob/e277f650148a95b35155e8a4ee93b2c1e56c5662/tools/perf/lib/figure/flat.py#L17)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

Any better?
</blockquote></details>

Much better
___
*[tools/perf/lib/figure/flat.py, line 17 at r4](https://reviewable.io/reviews/pmem/rpma/1366#-MpkvuAa3OBQqZrBHK-w:-MpkvuAa3OBQqZrBHK-x:b5fxonb) ([raw file](https://github.com/pmem/rpma/blob/96ecc00870b2241a355b2fc8fce7443aa0c4b54a/tools/perf/lib/figure/flat.py#L17)):*
> ```
> - a single element on this list represents exactly one output figure and
> - all elements on `figure['series']` lists represent exactly one series of
>   benchmarks.
> ```

2 x (`on the list` -> `in the list`)

**Review by ldorau — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366#-:-MpkxAV_Eso3HHR_PY8i:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @janekmi)

**Review by osalyk — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366#-:-MpkxTeGETiA3lJD5SbV:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @janekmi)

**Review by ldorau — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @janekmi)

**Review by grom72 — COMMENTED** [review]
> Reviewed 1 of 10 files at r2, 1 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1366)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**